packages/web-components/src/player-component/player-component.ts

```diff
@@ -483,10 +483,48 @@ export class PlayerComponent extends FASTElement {
         }
     }
 
+    private updateNextDayDisabled(): null {
+        const currentDayIndex = this.currentAllowedDays.findIndex(item => parseInt(item, 10) === this.currentDay);
+        if (currentDayIndex === this.currentAllowedDays.length - 1) {
+            const currentMonthIndex = this.currentAllowedMonths.findIndex(item => parseInt(item, 10) === this.currentMonth);
+            if (currentMonthIndex === this.currentAllowedMonths.length - 1) {
+                const currentYearIndex = this.currentAllowedYears.findIndex(item => parseInt(item, 10) === this.currentYear);
+                if (currentYearIndex === this.currentAllowedYears.length - 1) {
+                    this.player.disableNextDayButton(true);
+                    return null;
+                }
+            }
+        }
+        this.player.disableNextDayButton(false);
+        return null;
+    }
+
     private async selectNextDay() {
+        const currentDayIndex = this.currentAllowedDays.findIndex(item => parseInt(item, 10) === this.currentDay);
+        if (currentDayIndex === this.currentAllowedDays.length - 1) {
+            const currentMonthIndex = this.currentAllowedMonths.findIndex(item => parseInt(item, 10) === this.currentMonth);
+            if (currentMonthIndex === this.currentAllowedMonths.length - 1) {
+                const currentYearIndex = this.currentAllowedYears.findIndex(item => parseInt(item, 10) === this.currentYear);
+                this.currentYear = parseInt(this.currentAllowedYears[currentYearIndex + 1], 10);
```

Comment:
>this.currentYear = parseInt(this.currentAllowedYears[currentYearIndex + 1], 10); [](http://example.com/codeflow?start=16&length=80)

if this is the last year, return

Reply:
Added last year condition, the function will return in this condition.